src/andromede/model/constraint.py

```diff
@@ -84,3 +86,24 @@ def __init__(
                 self.upper_bound = upper_bound
             else:
                 self.upper_bound = literal(float("inf"))
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Constraint):
+            return False
+        return (
+            self.name == other.name
+            and expressions_equal(self.expression, other.expression)
+            and _expressions_equal_if_present(self.lower_bound, other.lower_bound)
+            and _expressions_equal_if_present(self.upper_bound, other.upper_bound)
+        )
+
+
+def _expressions_equal_if_present(
+    lhs: Optional[ExpressionNode], rhs: Optional[ExpressionNode]
+) -> bool:
```

Comment:
Since the same function is used in both constraint.py and variable.py, maybe move it to common.py ?